### ISSUE TITLE
File type comparisons only accept symbols despite documentation

### ISSUE THREAD

**Issue by stevendanna** [inspec/inspec#2204]
### Description

This issue was found by @Vasu1105 when working on https://github.com/chef/train/pull/193

The documentation here: https://www.inspec.io/docs/reference/resources/file/ states you can check the type of a file with code such as:

```
its('type') { should eq 'file' }
```

However, the `type` of a file is a symbol and it seems no coercion is done using either `eq` or `cmp`:

```
> cat test.rb
describe file("testfile") do
  its('type') { should eq 'file' }
  its('type') { should cmp 'file' }
  its('type') { should eq :file }
end
sdanna@thrace ~/oc/code/opscode/inspec (master %) > ls -l testfile
-rw-r--r--  1 sdanna  staff  0 Oct  3 11:54 testfile
sdanna@thrace ~/oc/code/opscode/inspec (master %) > be inspec exec test.rb
/Users/sdanna/.chefdk/gem/ruby/2.4.0/gems/rb-readline-0.5.3/lib/rbreadline.rb:13: warning: constant ::Fixnum is deprecated

Profile: tests from test.rb
Version: (not specified)
Target:  local://


  File testfile
     ∅  type should eq "file"

     expected: "file"
          got: :file

     (compared using ==)

     Diff:
     @@ -1,2 +1,2 @@
     -"file"
     +:file

     ∅  type should cmp == "file"

     expected: "file"
          got: file

     (compared using `cmp` matcher)

     ✔  type should eq :file

Test Summary: 1 successful, 2 failures, 0 skipped
```

**Comment by adamleff** [comment]
Thanks for the find, @stevendanna and @Vasu1105! We should probably override the `type` method in the InSpec `file` resource to `to_s` what we get back from Train, and then perhaps consider enhancing the `cmp` matcher to be crafty with symbols vs. strings. Either way, I personally think the docs are the "right way" and we just need to make the resource better.

**Comment by Vasu1105** [comment]
I have come across one more issue in relation to `type` matcher.. 
The Documentation here : https://www.inspec.io/docs/reference/resources/file/ here states `c (the file is a character device); use 'character to test for this file type`
```its('type') { should eq 'character' }```
However it fails since the value which compared this with is :character_device
```
×  File /dev/zero type should eq "character"
     
     expected: "character"
          got: :character_device
```
Same is the case with block device.

**Comment by Vasu1105** [comment]
Few more file resource bug
```
×  File /home/ubuntu/hello.txt mtime should eq "October 31 2015 12:10:45"
     
     expected: "October 31 2015 12:10:45"
          got: 1506423173
```

**Comment by adamleff** [comment]
PR opened to fix docs for block_device, character_device, type (which should use `cmp` to ensure string-to-symbol coercion) and add symbol/string coercion to the `cmp` matcher

**Comment by adamleff** [comment]
All fixes (most of which were documentation fixes) have been added to PR #2224
